scripts/run_pahfit

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env python
+
+import os
+import pkg_resources
+import argparse
+
+import matplotlib.pyplot as plt
+import matplotlib as mpl
+
+import astropy.units as u
+from astropy.table import Table
+from astropy.modeling.fitting import LevMarLSQFitter
+
+from pahfit.base import PAHFITBase
+
+
+def initialize_parser():
+    """
+    Command line parser for PAHFIT
+
+    Returns
+    -------
+    parser : argparse object
+    """
+    plottypes = ['png', 'jpg', 'jpeg', 'pdf', 'ps', 'eps', 'rgba',
+                 'svg', 'tiff', 'tif', 'pgf', 'svgz', 'raw']
+    savetypes = ['fits', 'votable', 'ipac', 'ascii.ecsv']
+    parser = argparse.ArgumentParser()
+    parser.add_argument('spectrumfile',
+                        help='name of file with observed spectrum')
+    parser.add_argument('packfile', help='name of PAHFIT pack file')
+    parser.add_argument('--savefig', action='store',
+                        default='pdf', choices=plottypes,
+                        help='Save figure to a file of specified type')
+    parser.add_argument('--showplot', action='store_true',
+                        help='display plot to the screen')
+    parser.add_argument('--saveoutput', action='store',
+                        default='ipac', choices=savetypes,
+                        help='Save fit results to a file of specified type')
+
+    return parser
+
+
+if __name__ == '__main__':
+
+    # setup and parse the command line
+    parser = initialize_parser()
+    args = parser.parse_args()
+
+    # read in the observed spectrum
+    # assumed to be astropy table compatibile and include units
+    specfile = args.spectrumfile
+    outputname = specfile.split('.')[0]
+    if not os.path.isfile(specfile):
+        pack_path = pkg_resources.resource_filename('pahfit', 'data/')
+        test_specfile = '{}/{}'.format(pack_path, specfile)
+        if os.path.isfile(test_specfile):
+            specfile = test_specfile
+        else:
+            raise ValueError(
+                'Input spectrumfile {} not found'.format(specfile))
+
+    # get the table format (from extension of filename)
+    tformat = specfile.split('.')[1]
+    obs_spectrum = Table.read(specfile, format=tformat)
+    obs_x = obs_spectrum['wavelength'].to(u.micron,
+                                          equivalencies=u.spectral())
+    obs_y = obs_spectrum['flux'].to(u.Jy,
+                                    equivalencies=u.spectral_density(obs_x))
+    obs_unc = obs_spectrum['sigma'].to(u.Jy,
+                                       equivalencies=u.spectral_density(obs_x))
+
```

Comment:
Are the units from the ipac file read and assigned to the the arrays from Table.read? If not, when converting the units in lines 66-71 wouldn't it raise an error ?

Reply:
Yes.  If there are not units, then this will cause an error.  Enforcing designation of units is something we can decide to do.  Having default units can be a bit error prone.  So, maybe we just say you have specify them?

Reply:
At least that is what I'd advocate.  Adding units to an IPAC file is straightforward, just including them in the column header info as strings.